File: challenges/001/descendingOrder.js
function descendingOrder(number){
    const array = [];

    for (const digit of String(n)) {
        const arrayItem = Number(digit)
        array.push(arrayItem)
    }

    // Iterate through the array
    for (let i = 0; i < array.length; i++) {
        // Assume the first digit is the largest
        let maxIndex = i;

        // Find the largest number and set it to max
        for (let j = i + 1; j < array.length; j++) {

            if (array[j] > array[maxIndex]) {
                maxIndex = j;
            }
        }

        [array[i], array[maxIndex]] = [array[maxIndex], array[i]]
    }

    return Number(array.join(''))
}

File: challenges/001/descending_order.py
def descendingOrder(number):
    # convert the int into a str
    numbers_str = str(number)

    # initialize an empty array
    numbers_list = []

    # iterate through the string and add to array
    for x in numbers_str: 
        numbers_list.append(x)
    

    # Iterate through the array
    for i in range(len(numbers_list)):

        # Find the largest number and set it to max
        for j in range(i + 1, len(numbers_list)):

            if numbers_list[j] > numbers_list[i]: 
                numbers_list[i], numbers_list[j] =  numbers_list[j], numbers_list[i]
               
    result = int(''.join(numbers_list))
    return result


print(descendingOrder(42145))


"""
More pythonic version:

def descendingOrder(number): 
    return int(''.join(sorted(str(number), reverse=True)))
"""